android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFirebaseCoreService.java

```diff
@@ -86,14 +85,14 @@ public ScopedFirebaseCoreService(Context context, RawManifest manifest, Experien
     updateFirebaseApp(mAppOptions, mAppName);
   }
 
-  private static String getEncodedExperienceId(ExperienceId experienceId) {
+  private static String getEncodedExperienceId(ExperienceKey experienceKey) {
     try {
-      String encodedUrl = experienceId.getUrlEncoded();
+      String encodedUrl = experienceKey.getUrlEncodedStableLegacyId();
       byte[] data = encodedUrl.getBytes("UTF-8");
       String base64 = Base64.encodeToString(data, Base64.URL_SAFE | Base64.NO_PADDING | Base64.NO_WRAP);
       return base64;
     } catch (UnsupportedEncodingException e) {
-      return Integer.toString(experienceId.hashCode());
+      return Integer.toString(experienceKey.getStableLegacyId().hashCode());
     }
   }
```

Comment:
I don't want to hold up this whole PR but I think we should have a focused discussion about what ID is correct to use here, assuming we phase out full names for new projects.